lib/crypto/sr25519/sr25519.go

```diff
@@ -43,6 +44,25 @@ type PrivateKey struct {
 	key *sr25519.SecretKey
 }
 
+// VerifySignature verifies a signature given a public key and a message
+func VerifySignature(publicKey, signature, message []byte) (bool, error) {
+	pubKey, err := NewPublicKey(publicKey)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch sr25519 public key: %w", err)
```

Comment:
Are we actually fetching a public key? Sorry I have no idea what `NewPublicKey` does

Reply:
`ed25519.NewPublicKey` and `sr25519.NewPublicKey` functions creates a public key given a byte slice, @1garo I think we can just return the `err`

Reply:
Yes let's return `err` only :+1: 

The `fetch` is wrong here, and the `sr25519` error context should be added by the caller. I'll add comment on the caller.